st2common/st2common/models/system/action.py

```diff
@@ -544,6 +544,15 @@ def _execute_remote_command(self, command):
 
         if output.failed:
             msg = 'Remote command %s failed.' % command
+            # XXX: Note Fabric doesn't handle unicode correctly if stdout or stderr contains
+            # unicode and action fails. For now, just log stdout and stderr so we can debug
+            # from logs.
+            # Fabric will show an exception traceback like:
+            # 'ascii' codec can't encode character u'\u2018' in position 93:
+            # ordinal not in range(128)
+            #
+            LOG.error('stderr: %s', output.stderr)
+            LOG.error('stdout: %s', output.stdout)
             LOG.error(msg)
             raise Exception(msg)
```

Comment:
Ah, I see.

We throw and that's why stdout and stderr is not included in the output. Imo it would be good and useful to include it. We could throw a special exception and handle this case in the fabric runner.